src/providers/EtherspotContextProvider.tsx

```diff
@@ -276,13 +284,76 @@ const EtherspotContextProvider = ({
     return [];
   }, [sdk, accountAddress]);
 
+  const getSmartWalletBalancesPerChain = useCallback(async (walletAddress: string, supportedChains: Chain[]) => {
+    if (!sdk || !walletAddress) return;
+    let balanceByChain : IBalanceByChain[] = []
+    for (let index = 0; index < supportedChains.length; index++) {
+      const element = supportedChains[index];
+      try {
+        let supportedAssets = await getSupportedAssetsWithBalancesForChainId(
+          element.chainId,
+          true,
+          walletAddress,
+          false
+        );
+        balanceByChain.push({
+          title: element.title,
+          chain: element.chainId,
+          total: Number(supportedAssets.reduce((acc,curr)=>{
```

Comment:
can also rename `acc, curr` into something more relatable?
```suggestion
          total: Number(supportedAssets.reduce((acc, curr)=>{
```